community/kernel/src/main/java/org/neo4j/kernel/impl/proc/ProcedureGDBFacadeSPI.java

```diff
@@ -205,6 +184,18 @@ public void shutdown()
     @Override
     public KernelTransaction beginTransaction( KernelTransaction.Type type, SecurityContext securityContext, long timeout )
     {
-        throw new UnsupportedOperationException();
+        try
+        {
+            availability.assertDatabaseAvailable();
+            KernelTransaction kernelTx = sourceModule.kernelAPI.get().newTransaction( type, this.securityContext, timeout );
```

Comment:
Perhaps this makes it safe? Using this.securityContext. What about also renaming the method parameter to 'ignoredSecurityContext' to make the code look a bit safer?